src/Microsoft.ML.StandardLearners/Standard/Online/OnlineLinear.cs

```diff
@@ -42,8 +42,10 @@ public abstract class OnlineLinearArguments : LearnerInputBaseWithLabel
         [Argument(ArgumentType.AtMostOnce, HelpText = "Size of cache when trained in Scope", ShortName = "cache")]
         public int StreamingCacheSize = 1000000;
 
+        [BestFriend]
         internal class OnlineDefaultArgs
         {
+            [BestFriend]
             internal const int NumIterations = 1;
```

Comment:
Why not making this public? If the outer class is internal, that should be sufficient right?

Reply:
Same question applies everywhere else you have argument defaults

---
In reply to: [243470421](https://github.com/dotnet/machinelearning/pull/1930#discussion_r243470421) [](ancestors = 243470421)

Reply:
Hi @abgoswam, incidentally this issue raised by @artidoro is the one thing that is preventing me from signing off. While the mistake of whoever wrote these default classes (making members of an *internal* class itself internal, which is utterly pointless and counterproductive) was a regrettable mistake, by sprinkling this attribute everywhere instead of fixing the original problem (that is, these members should have the public access modifier), we are making the code appreciably more difficult to read. While I know you did not commit that original error, since you're now relying on them we should fix it now.

Reply:
makes sense. thanks for the details. 

---
In reply to: [243577623](https://github.com/dotnet/machinelearning/pull/1930#discussion_r243577623) [](ancestors = 243577623)